### PR TITLE
add backend and tests for resolvers

### PR DESCRIPTION
**Pull request by pb82**
The backend part for the resolver mappings.

### CODE REVIEW

File: server/gql/schema.js [hunk 1]
```diff
@@ -19,19 +19,38 @@ const Schema = buildSchema(`
         deleteDataSource(id: Int!): DataSource
         updateDataSource(id: Int!, name: String!, type: DataSourceType!, config: String!): DataSource
         updateSchema(id: Int!, schema: String!): Schema
+        upsertResolver(
+            id: Int
```
Comment (josemigallas) [inline server/gql/schema.js]: What's this id?

File: server/gql/schema.js [hunk 2]
```diff
@@ -19,19 +19,38 @@ const Schema = buildSchema(`
...
+            schemaId: Int!
+            dataSourceId: Int! 
+            type: String!,
+            field: String!
+            requestMapping: String!
+            responseMapping: String!
+        ): Resolver
     },  
-    type DataSource {
+    type DataSource {
```
Comment (josemigallas) [inline server/gql/schema.js]: Additional spaces

File: server/gql/schema.js [hunk 3]
```diff
@@ -56,6 +75,37 @@ const listDataSources = ({ name }) => {
...
+    }
+});
+
+const upsertResolver = async ({
+    id, schemaId, dataSourceId, type, field,
+    requestMapping, responseMapping
+}) => {
+    const properties = {
+        GraphQLSchemaId: schemaId,
+        DataSourceId: dataSourceId,
```
Comment (josemigallas) [inline server/gql/schema.js]: Is `properties.DataSourceId` upper case on purpose?

File: server/gql/schema.js (hunk 1 above)
Comment (pb82) [reply server/gql/schema.js]: The id of the resolver. If provided it triggers an update, otherwise an insert.

File: server/gql/schema.js (hunk 3 above)
Comment (pb82) [reply server/gql/schema.js]: Yes, thats how Sequelize expects them.

Comment (pb82) [reply server/gql/schema.js]: Could be overridden i guess. There is also a different API that we can use to add a resolver to a schema and data source but that would require us to fetch those instances first.

File: server/gql/schema.js (hunk 1 above)
Comment (josemigallas) [reply server/gql/schema.js]: Maybe a silly question but doesn't that usually depend on the datasource id?

Comment (pb82) [reply server/gql/schema.js]: No, Resolver and Data Source are two different things. They are associated (one to many: one data source cab be used by many resolvers) but different tables and GraphQl entities.

**Pull request merged**